opendatahub-io/odh-dashboard
# Adds support for TrustyAI Operator

Closes: #1276 

## Description


* Adds TrustyAI client capabilities for interacting with the operator and managing the service lifecycle. [non-visual]
* Adds Settings tab to project details page with a model bias section allowing for installing / uninstalling TrustyAI.
![localhost_4010_projects_test1](https://github.com/opendatahub-io/odh-dashboard/assets/4092230/2a70d749-721e-499a-9819-83eccd7517a4)
![localhost_4010_projects_test1 (1)](https://github.com/opendatahub-io/odh-dashboard/assets/4092230/44214a4c-703c-4a97-a78e-04e58c49b849)
![localhost_4010_projects_test1 (2)](https://github.com/opendatahub-io/odh-dashboard/assets/4092230/6dc142ad-00fd-424c-a5f1-27c0e6f189e3)

## How Has This Been Tested?




Test 1: Verify TrustyAI CR creation and deletion
1) Create a new Data Science Project
2) Goto the Details -> Settings tab
3) Check the "Enable TrustyAI" checkbox
4) Verify you receive a success message and the checkbox is now ticked.
5) In the OpenShift console check that a `trustyai-service` CR has been installed in your namespace
6) Now check the Deployments tab for a trusty-service in your namespace.
7) Go back to the Dashboard and untick the checkbox
8) Verify a confirm modal appears, and confirm the delete
9) Verify the checkbox is unticked.
10) Go back to the OpenShift console and confirm the CR and service (from step 5 and 6) have been removed. 

Test 2: Verify TrustyAI works correctly
1) Go to Model Serving Global page
2) Open model `demo-loan-nn-onnx-alpha`
3) Verify the Model Bias tab appears.
4) Verify creating a Bias Configuration works
5) Verify the new configuration is selectable on the Charts view - NOTE: there will be no data for the charts as there is currently a bug in the TrustyAI service.
6) Verify Bias configuration duplicate and delete functionality works

Test 3: Verify Bias Tab is not visible when TrustyAI service is not installed
1) Goto project settings for `trustyai-e2e-modelmesh`
2) Uninstall the TrustyAI service.
3) Go back to the `demo-loan-nn-onnx-alpha` model serving screen
4) Verify the Model Bias tab is not visible


## Test Impact


TBD

## Request review criteria:
